lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,94 @@
+package handler
+
+import (
+	"context"
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+	"log"
+)
+
+type Handler struct {
+	betRepository CalcBetRepository
+}
+
+func New(betRepository CalcBetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+func (h *Handler) HandleBets(ctx context.Context, bets <-chan rabbitmqmodels.Bet ) {
+	go func() {
+		for bet := range bets {
+			log.Println("Processing bet, betId:", bet.Id, " with selectionId:", bet.SelectionId)
+
+			domainBet := domainmodels.Bet{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			err := h.betRepository.InsertCalcBet(ctx, domainBet)
+			if err != nil {
+				log.Println("Failed to insert bet, error: ", err)
+				continue
+			}
+			log.Println("Inserted bet, betId: ", domainBet.Id, ", selectionId: ", domainBet.SelectionId)
+
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				continue
+			}
+		}
+	}()
+}
+
+func (h *Handler) HandleEventUpdates(
+	ctx context.Context,
+	eventUpdates <-chan rabbitmqmodels.EventUpdate,
+	) <-chan rabbitmqmodels.BetCalculated {
+	betsCalculated := make(chan rabbitmqmodels.BetCalculated)
+
+	go func() {
+		defer close(betsCalculated)
+
+		for eventUpdate := range eventUpdates {
+			log.Println("Processing event update, Id:", eventUpdate.Id)
+
+			domainBets, exists, err := h.betRepository.GetCalcBetsBySelectionID(ctx, eventUpdate.Id)
+			if err != nil {
+				log.Println("Failed to fetch a bet which should be updated, error: ", err)
+				continue
+			}
+			if !exists {
+				log.Println("A bet which should be updated does not exist, selection id: ", eventUpdate.Id)
```

Comment:
This message assumes that there should be bets for the `eventUpdate.Id`.

It is completely fine that there are no bets for the mentioned event.
Something like: `event update <x> references 0 bets` could be more appropriate.